Services/UI/OJS.Services.Ui.Business/Implementations/SearchBusinessService.cs

```diff
@@ -41,52 +41,86 @@ public class SearchBusinessService : ISearchBusinessService
             throw new BusinessServiceException(validationResult.Message);
         }
 
-        var searchListingModel = new SearchForListingServiceModel();
-        model.ItemsPerPage = DefaultItemsPerPage;
+        var contestSearchListingModel = new ContestSearchForListingServiceModel();
+        await this.PopulateSelectedConditionValues(model, contestSearchListingModel);
+
+        var modelResult = model.Map<PagedResult<ContestSearchForListingServiceModel>>();
+        modelResult.Items = new[] { contestSearchListingModel, };
+
+        return modelResult;
+    }
+
+    public async Task<PagedResult<ProblemSearchForListingServiceModel>> GetProblemSearchResults(
+        SearchServiceModel model)
+    {
+        NormalizeSearchModel(model);
+
+        var validationResult = this.searchValidationService.GetValidationResult(model.SearchTerm);
+
+        if (!validationResult.IsValid)
+        {
+            throw new BusinessServiceException(validationResult.Message);
+        }
+
+        var problemsSearchListingModel = new ProblemSearchForListingServiceModel();
+        await this.PopulateSelectedConditionValues(model, problemsSearchListingModel);
+
+        var modelResult = model.Map<PagedResult<ProblemSearchForListingServiceModel>>();
+        modelResult.Items = new[] { problemsSearchListingModel, };
```

Comment:
Why put `problemsSearchListingModel` in an array?

Reply:
Destructing values from fetched data has been refactored as well as the way data is actually being fetched in the BE.